tests/mapper/test_adjacency_generators.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from stellargraph.core.utils import normalize_adj
+from stellargraph.mapper.adjacency_generators import AdjacencyPowerGenerator
+from .test_graphwave_generator import barbell
+
+
+import numpy as np
+import pytest
+
+
+def test_init(barbell):
+
+    generator = AdjacencyPowerGenerator(barbell, num_powers=5)
+    num_nodes = len(barbell.nodes())
+
+    assert generator.num_powers == 5
+    assert generator.Aadj_T.shape == (num_nodes, num_nodes)
+    assert generator.transition_matrix_T.shape == (num_nodes, num_nodes)
+
+
+def test_bad_init(barbell):
+
+    with pytest.raises(TypeError):
+        generator = AdjacencyPowerGenerator(None, num_powers=5)
+
+    with pytest.raises(ValueError, match="num_powers: expected.*found -1"):
+        generator = AdjacencyPowerGenerator(barbell, num_powers=-1)
+
+    with pytest.raises(TypeError, match="num_powers: expected.*found float"):
+        generator = AdjacencyPowerGenerator(barbell, num_powers=1.0)
+
+
+def test_flow(barbell):
+
+    generator = AdjacencyPowerGenerator(barbell, num_powers=5)
+    for i, x in enumerate(generator.flow(batch_size=1)):
+
+        if i > 5:
+            break
+
+    assert i > 5
```

Comment:
What's this test trying to catch? Maybe a way to rephrase it would be to just check the length of the sequence object matches expectation, if that was the intention.

Reply:
this is trying to test whether the `Dataset` loops infinitely. The graph only has 5 nodes so if `i > 5` it means that the `Dataset` is repeated. 

I looked into it and there's a `tf.data.experimental.cardinality` function that I'll use instead of looping through 👍